tests/test_merge.py

```diff
@@ -51,33 +52,311 @@ def test_merge_branch(wst, capsys):
     assert out.split('\n')[0] == 'Merging 3.0.x into master'
 
 
-def test_merge_branch_with_whitelist(wst, capsys):
+def test_merge_branch_skip_last(wst, capsys):
     config.merge.branches = '1.0.x 2.0.x 3.0.x master'
+    with temp_git_repo():
+        # Dummy commit
+        run('git commit --allow-empty -m dummy-commit')
+
+        run('git checkout -b 3.0.x')
+
+        # commit 1
+        run('touch temp1.xml')
+        run('git add -A')
+        run('git commit -m commit1')
+        commit1 = run('git rev-parse --verify HEAD', return_output=True)
+
+        # commit 2
+        run('touch temp2.xml')
+        run('git add -A')
+        run('git commit -m  commit2')
+        commit2 = run('git rev-parse --verify HEAD', return_output=True)
+
+        # commit 3
+        run('touch temp3.xml')
+        run('git add -A')
+        run('git commit -m  [skip]')
+        skipped = run('git rev-parse --verify HEAD', return_output=True)
 
+        run('git checkout master')
+
+        run('touch temp4.xml')
+        run('git add -A')
+        run('git commit -m commit4')
+        commit4 = run('git rev-parse --verify HEAD', return_output=True)
+
+        wst('merge 3.0.x --skip-commits \'skip\' \'space separated example\'')
+
+        changes = run('git log --oneline', return_output=True)
+        # Change should have been in the log entry
```

Comment:
1. maybe update this comment to be a little more descriptive about what we are checking
2. Instead of just blindly searching the entire string for the presence of our commits can we go through each line of the commit and make sure it is exactly what we expect. maybe it makes the test a little more brittle but i worry that some of these assertions could be true even though we don't get what we expect.